contracts/Pricing.sol

```diff
@@ -76,33 +76,42 @@ contract Pricing is IPricing {
      */
     function recordTrade(uint256 tradePrice, uint256 fillAmount) external override onlyTracer {
         uint256 currentOraclePrice = oracle.latestAnswer();
+        // update rates if a trade has not been recorded in the last hour
         if (startLastHour <= block.timestamp - 1 hours) {
-            // emit the old hourly average
+            // get the last recorded hourly price, returns max integer if no trades occurred
             uint256 hourlyTracerPrice = getHourlyAvgTracerPrice(currentHour);
-            emit HourlyPriceUpdated(hourlyTracerPrice, currentHour);
 
-            // update funding rate for the previous hour
-            updateFundingRate();
+            // emit the hourly average and udpate funding rate if trades occurred
+            if (hourlyTracerPrice != type(uint256).max) {
+                // emit the old hourly average
+                emit HourlyPriceUpdated(hourlyTracerPrice, currentHour);
+
+                // update funding rate for the previous hour
+                updateFundingRate();
+            }
+
+            uint256 elapsedHours = (block.timestamp - startLastHour) / 3600;
+            // if more than one hour passed, update any skipped hour prices as 0
+            if (elapsedHours > 1) {
+                for (uint256 i = 0; i < elapsedHours - 1; i++) {
+                    currentHour = (currentHour + 1) % 24;
+                    updatePrice(0, 0, 0, true);
```

Comment:
I wonder if this is the most efficient way to do this -- this is making a change to a global state variable at every turn even though you know what it is going to be at the end. Maybe it be worth making `updatePrice` use a pure function in the if-bodies or something, and reuse that function here?

Reply:
Updated function to take in an hour variable, then used a local variable `staleHour` in for loop to address this.